components/nav/NavTitle.vue

```diff
@@ -2,6 +2,14 @@
 const { env } = useBuildInfo()
 const router = useRouter()
 const back = ref<any>('')
+
+const nuxtApp = useNuxtApp()
+
+const onClickLogo = () => {
+  // @ts-expect-error missing types
+  nuxtApp.hooks.callHook('elk-logo:click')
```

Comment:
You could probably add an ambiant type to this no?

Reply:
Also, I'm not pretty sure using a nuxt hook is the best move..

Reply:
That's said: you can extends `RuntimeNuxtHooks` interface to add the new one and remove ts ignore ;)

Reply:
I think adding an ambient type is better than using ts-expect-error , but I'm not sure how to extend `RuntimeNuxtHooks`

I did this but it doesn't seem to work.

```ts
// shims.d.ts

declare module '@nuxt/schema' {
  interface RuntimeNuxtHooks {
    'elk-logo:click': () => void
  }
}

export {}
```

Reply:
I had a similar functional requirement in a company project, currently I solved it with hookable, but that was a vue project not nuxt.

I'm also not sure if the nuxt core team expects hookable to be used in this way, but so far it seems to work.

Reply:
Ask this in the #nuxt discord channel, Daniel is usually answer quickly.

Reply:
It is resolved at c78b4312a51a2d13b576e0e70fee9c290c7cb2af